core/src/main/clojure/xtdb/log.clj

```diff
@@ -196,17 +196,14 @@
       (.writeObject xtql-op-writer (ClojureForm. op))
       (.endStruct xtql-writer))))
 
-(defn encode-sql-args [^BufferAllocator allocator, query, arg-rows]
-  (let [plan (sql/compile-query query)
-        {:keys [^long param-count]} (meta plan)
-
-        vecs (ArrayList. param-count)]
+(defn encode-sql-args [^BufferAllocator allocator, arg-rows]
+  (let [max-param-count (apply max (map count arg-rows))
+        vecs (ArrayList. max-param-count)]
     (try
-      ;; TODO check arg count in each row, handle error
```

Comment:
If indeed we do see have a mismatch here expecting to throw an error at a later point when compiling the query.

Reply:
@danmason could you pop a test in to confirm this?